h/search/query.py

```diff
@@ -109,6 +120,24 @@ def extract_sort(params):
             "unmapped_type": "boolean",
         }
     }]
+    search_after = params.pop("start", None)
+    if search_after:
+        if sort_by in ["updated", "created"]:
+            return (_date_parser(search_after), sort)
+        else:
+            return ([search_after], sort)
+    return ([], sort)
+
+
+def _date_parser(str_value):
+    try:
+        return [float(str_value)]
+    except ValueError:
+        try:
+            date = parse(str_value)
+            return [(date - dt.utcfromtimestamp(0)).total_seconds() * 1000.0]
```

Comment:
When I tried to use the API locally in the most obvious way, by taking the value of "updated" from the last annotation in the search results ("2018-08-27T11:02:10.133597+00:00") and passing that as the "start" value for the next call, I got this error:

```
  File &#34;/Users/robert/hypothesis/repos/h/h/views/api/annotations.py&#34;, line 104, in search
    stats=stats).run(params)
  File &#34;/Users/robert/hypothesis/repos/h/h/search/core.py&#34;, line 56, in run
    total, annotation_ids, aggregations = self._search_annotations(params)
  File &#34;/Users/robert/hypothesis/repos/h/h/search/core.py&#34;, line 88, in _search_annotations
    body=self.builder.build(params))
  File &#34;/Users/robert/hypothesis/repos/h/h/search/query.py&#34;, line 47, in build
    p_search_after, p_sort = extract_search_after_sort(params)
  File &#34;/Users/robert/hypothesis/repos/h/h/search/query.py&#34;, line 126, in extract_search_after_sort
    return (_date_parser(search_after), sort)
  File &#34;/Users/robert/hypothesis/repos/h/h/search/query.py&#34;, line 138, in _date_parser
    return [(date - dt.utcfromtimestamp(0)).total_seconds() * 1000.0]
TypeError: can&#39;t subtract offset-naive and offset-aware datetimes
```

`date.timestamp()` here will return the POSIX timestamp as a float in seconds, which you can then multiply to convert it to be in ms. The user may pass in a timestamp with or without a timezone. If they omit the timezone then the server, which should always have the system clock set to UTC, will assume UTC. If there is a way to enforce this assumption that would be even better.